### PR TITLE
Correct rust code block in *Dataflow Analysis*

### PR DESCRIPTION
**Pull request by xFrednet**
While reading the guide, I noticed that the code example didn't compile. This change, simply corrects the code and adds syntax highlighting by marking the code block as rust code.

### CODE REVIEW

File: src/mir/dataflow.md
Comment (camelid) [inline src/mir/dataflow.md:141]: Can you unindent the body of the function since we're hiding the function header anyway?

**Comment by camelid** [comment]
There's also a link-checking error (not your fault; it's just because the file hasn't been changed and thus checked in a while). Do you mind fixing it? Thanks!

```
error: It looks like "mir/debugging.md" wasn't included in SUMMARY.md
    ┌─ mir/dataflow.md:217:1
    │
217 │ [Debugging MIR]. Start with `-Z dump-mir=F -Z dump-mir-dataflow`, where `F` is
    │ ^^^^^^^^^^^^^^^ It looks like "mir/debugging.md" wasn't included in SUMMARY.md

Error: One or more incorrect links
```

**Comment by xFrednet** [comment]
> Do you mind fixing it? Thanks!

I've tried fixing it, but don't really understand the problem. The error mentions a file which is included in the summary. I've now update the summary to use the same title and the lint to point to the `.md` file, but the error still persists.

This could be https://github.com/Michael-F-Bryan/mdbook-linkcheck/issues/77 . Do you have an idea how to fix the error?

**Review by camelid — APPROVED** [review]
Thanks! It does look like a bug in linkcheck like you said, and since this change certainly doesn't cause the issue, we'll ignore it. It'd be good to get to the bottom of this though.

**Comment by camelid** [comment]
Argh we can't merge until CI is fixed sigh.

**Comment by camelid** [comment]
Will merge once someone else from @rust-lang/wg-rustc-dev-guide approves my CI change.

**Comment by camelid** [comment]
Received the [go-ahead](https://rust-lang.zulipchat.com/#narrow/stream/196385-t-compiler.2Fwg-rustc-dev-guide/topic/linkcheck.20bugs/near/457137498) on Zulip.

**Pull request merged**
